stop_and_wait_plugin/src/stop_and_wait_plugin.cpp

```diff
@@ -162,12 +162,22 @@ namespace stop_and_wait_plugin
 
             }
            
-            double ending_downtrack = stop_and_wait_maneuver.end_dist;
-            maneuver_time_ = ros::Duration(stop_and_wait_maneuver.end_time - stop_and_wait_maneuver.start_time).toSec();
+            double ending_downtrack = stop_and_wait_maneuver.end_dist; 
+            double start_speed = current_speed_;    //Get static value of current speed at start of planning
+            //maneuver_time_ = ros::Duration(stop_and_wait_maneuver.end_time - stop_and_wait_maneuver.start_time).toSec();
             
+            maneuver_time_ = (3*(ending_downtrack - starting_downtrack))/(2*start_speed);
+
             double delta_time, curr_time;
-            if(current_speed_ == 0 && starting_downtrack >= ending_downtrack)  //If at end_dist return zero speed trajectory
+            if(start_speed == 0 )  //If at end_dist return zero speed trajectory
```

Comment:
Same comment as earlier with regards to doubles, generally speaking comparing doubles using `==` is a no-no given the quirks of how they're stored. You can have cases where if `x` is a double `(x * 2)/2) == x` is false and other sorts of weird issues which end up being very challenging to debug when they come up.